docs/pages/config/reference.mdx

```diff
@@ -121,13 +121,23 @@ The following options are available in both single- and multiple-namespace modes
     </Params>
 
   </Param>
+
   <Param name="modules">
+    <a id="modules" />  
     A list of modules to install into the world during the deploy step.
 
     <Params title="Module config options">
       <Param name="artifactPath">Relative path to the module's compiled JSON artifact (usually in `out`) or an import path if using a module from an npm package. This path is resolved using [Node's module `require` API](https://nodejs.org/api/modules.html#modulerequireid).</Param>
+
       <Param name="root">Whether or not to install this as a root module. Defaults to `false`.</Param>
-      <Param name="args">A list of arguments used to call the module's install function.</Param>
+
+      <Param name="args">
+        A list of arguments used to call the module's install function.
+        Each argument is a structure with two fields:
+        - **type** - Solidity data type.
+        - **value** - the value.
+          To encode a complex data type, such as a structure or an array, you can use Viem's [`encodeAbiParameters`](https://viem.sh/docs/abi/encodeAbiParameters.html).
+      </Param>
```

Comment:
Can we use embedded params rather than a list? see system deploy section as an example

Reply:
I tried that first. But it doesn't give enough space for the text at the third level of nesting, IMAO.

![Screenshot 2024-08-27 at 9 55 07 AM](https://github.com/user-attachments/assets/29532847-fdfe-4890-aaa3-aa0960f4e245)